internal/controllers/clusterextension_controller.go

```diff
@@ -206,6 +207,11 @@ func (r *ClusterExtensionReconciler) reconcile(ctx context.Context, ext *ocv1.Cl
 	installedBundle, err := r.InstalledBundleGetter.GetInstalledBundle(ctx, ext)
 	if err != nil {
 		setInstallStatus(ext, nil)
+		var saerr *tokengetter.SANotFoundError
+		if errors.As(err, &saerr) {
+			l.Info("is a SAError")
+			err = saerr
+		}
```

Comment:
I think there are a lot of other places where this error could be returned, likely anywhere we are doing interactions with the cluster since we are using the provided ServiceAccount for those interactions. Off the top of my head, that would be:
- Getting the installed bundle (here)
- Applying contents to the cluster 
- Setting up caches and informers for managed content

Is there a way we could more effectively capture this error type when setting the status conditions we are wanting a more specific message for?

Reply:
I understand how this is working in this particular error check, but I think there are other error checks in which it is possible for this error type to be returned as part of the wrapped error chain. Some off the top of my head is:
- When we attempt to fetch the currently installed bundle (which you already cover with this check here)
- When we attempt to apply content to the cluster (not covered by your changes)
- When we attempt to create informers to manage the content owned by a ClusterExtension (not covered by your changes)

The second piece of my comment was an attempt to get at the fact that instead of repeating this across all the error scenarios, and since we really only care about the message that is set in some of the status conditions when receiving this error, we could probably put this error check in the logic that is used to set the conditions we want to "customize" the errors for.

Reply:
will work on 
1) unit testcases
2) encompassing more scenarios:
```
When we attempt to apply content to the cluster (not covered by your changes)
When we attempt to create informers to manage the content owned by a ClusterExtension (not covered by your changes)